In [2]:
#from IPython.display import Image
#Image("output/piechart.png")
import pandas as pd
dog_breed_stats_df = pd.read_csv('dog_breed_stats.csv', index_col = 0)
dog_stages_stats_df = pd.read_csv('dog_stages_stats.csv', index_col = 0)
weekday_stats_df = pd.read_csv('weekday_stats_df.csv', index_col = 0)
twitter_archive_master = pd.read_csv('twitter_archive_master.csv')

# Data Analytics and Visualization

After the whole Gathering, Assessing and Claning process we end up with a dataset with 2142 observations. These are tweets from the WeRateDogs twitter account (https://twitter.com/dog_rates).

The most important Information gathered was kept on a tidy concise table. The author made an effort to remove all tweets not related to dogs. All reteweets were removed, just original tweets were kept on the dataset. All tweets kept do have the special rating system offered by the page author, http://knowyourmeme.com/memes/theyre-good-dogs-brent.

We will now proceed with a short evaluation of the data gathered and offer some insights. 

## Most Twitted Dog Breed

Ignoring the observations which did not produce any dog breed (457 - 21.33% of all observations), we have the following counts on the Top 6:

    - Golden Retriever                  158
    - Labrador Retriever                108
    - Pembroke                           95
    - Chihuahua                          91
    - Pug                                62
    - Toy Poodle                         51

Golden Retrievers are Labrador Retrievers are definetely the most popular dogs on Twitter. They out number all other breeds by a lot. The author of this study agrees with this data, Retrievers are definitely the coolest dogs out there. (The author is biased). We have a total of 112 different Dog Breeds on our dataset. Bellow is shown the whole dataset of all our breed of dogs and their respective statistics.

In [3]:
dog_breed_stats_df.head()

count  count_per  favorite_count_sum  retweet_count_sum  \
Not Defined           457  21.335201             3409434            1228719   
Golden Retriever      158   7.376284             1946405             585506   
Labrador Retriever    108   5.042017             1262718             402389   
Pembroke               95   4.435107             1040459             290206   
Chihuahua              91   4.248366              772178             258789   

                    rating_num_mean  rating_den_mean  favorite_count_mean  \
Not Defined               14.954048        10.363239          7460.468271   
Golden Retriever          12.943038        11.265823         12319.018987   
Labrador Retriever        13.638889        12.222222         11691.833333   
Pembroke                  11.410526        10.000000         10952.200000   
Chihuahua                 10.549451        10.000000          8485.472527   

                    retweet_count_mean  
Not Defined                2688.663020  
Golden Retriever           3705.734177  
Labrador Retriever         3725.824074  
Pembroke                   3054.800000  
Chihuahua                  2843.835165

If we take a look at the Top 15 most tweeted dog breeds, we will find the following information exemplified in this pie chart. Golden Retrievers take a lead with 18% of Tweets from the Top 15. Eskimo Dog closes the Top 15 with a 2.5% possesion of the Tweets. The insigh is if you want your dog to be featured on the WeRateDogs Twitter account, make sure you own a Golden Retriever, since this will you the best change to do so.

![title](output/piechart.png)


We will not dive into a more detailed analysis of which breed is the most popular and more Tweeted, on absolute terms and average per number of tweets.

From the following two plots we can draw the following conclusion:
    
    - Being Retweeted more does not grant that it will be favorited/retweeted more
    - Case being the Breeds Samoyed, French Bulldog or Cocker Spaniel
    - These dog Breeds are very popular amongst people, so it is natural that they get more attention from users
    


![title](output/Top15favRT.png)

Taking that in consideration, let us look at the Mean Favorite and Retweet counts. It can clearly be seen the observation made before.

    - The French Bulldog is the most popular dog amongst users.
    - They only represent 3.6% of the Author's Tweets, but they tend to be favorited/retweeted the most.

![title](output/Top15Means.png)

Although the rating give by the author is not very accurate or a measurable unit, we can still show it here, since it is one of the best features of this Tweet page.

    - The Author of the page tends to give the best ratings to the Breed Chow
    - Although it also gives it a higher denominator, so the ratio is kept
    - It seems that the Pomeranian has the best rating

![title](output/Top15RatMeans.png)

Bellow it is shown almost the same information has before, but now sorted by the Mean Favorited Dog Breed.
    
    - The French Bulldog, has seen, is the best Favorited Mean
    - Having the best Favorite Mean does not grant the best Retweet Mean
    - Same applies to the contrary relation (Seen bellow)

![title](output/Top15BreedMeanFav.png)

![title](output/Top15BreedMeanRT.png)

![title](output/Top15BreedMeanRating.png)

The following plots will show:
    
       - Sorted by Most Favorited Dog Breeds
       - Sorted by Most Retweeted Dog Breeds

![title](output/Top15MostFav.png)

![title](output/Top15MostRT.png)

## Most Popular Dog Breed

Taking in consideration all shown above, we can say that the most Popular Breeds are the following, in no particular order.

    - Samoyed                     
    - Eskimo Dog                  
    - Pembroke                    
    - Malamute                    
    - Labrador Retriever          
    - Chow                        
    - Golden Retriever            
    - Cocker Spaniel              
    - Toy Poodle                  
    - Chesapeake Bay Retriever    
    - French Bulldog              
    - Pomeranian                  
    - Chihuahua                   
    - Pug                         

These dog Breeds show up on every plot shown above, which have all sorts of different measureable analytics. If one takes in consideration just the average Favorited/Retweets, one has to say that the French Bulldog will be the most popula dog Breed amongst users.

# Analysis by Dog Stage (Puppo, Floofer, Doggo, Pupper)

In [3]:
dog_stages_stats_df

count  count_per  favorite_count_sum  retweet_count_sum  \
not_defined   1802  84.126984            15324184            4697720   
pupper         224  10.457516             1593158             543738   
doggo           84   3.921569             1472466             531492   
puppo           23   1.073763              502427             151511   
floofer          9   0.420168               98663              36898   

             rating_num_mean  rating_den_mean  favorite_count_mean  \
not_defined        12.402331        10.569367          8503.986681   
pupper             10.808036        10.000000          7112.312500   
doggo              11.761905        10.000000         17529.357143   
puppo              12.000000        10.000000         21844.652174   
floofer            11.888889        10.000000         10962.555556   

             retweet_count_mean  
not_defined         2606.947836  
pupper              2427.401786  
doggo               6327.285714  
puppo               6587.434783  
floofer             4099.777778

Let us now focus on influence of the dog stage on the statistics of the tweets. Unfortunately, we do no have a lot of observations to draw conclusions from, so this will be a rather limited analysis. Please refer to the bar graphs bellow. 

    - Puppos are definetely the most popular amongst users. On average they tend to be Favorited and Retweeted more than any other stage.

    - The owner of the page tends to give the same rating to all stages, a little bit less to puppers.

![title](output/DogStageMostMean.png)

If we look at the absolute numbers, we can see that:

    - The author is biased to post Puppers, or at least to refer to dogs has Puppers on the posts. 
            It far exceeds any other dog stage posted
    
    - Despite that fact, users tend to Favorite and Retweet almost equally Puppers and Doggos. 
            Puppos and Floofers get very little love from Tweeter users on absolute terms, but this is due tot he fact that there are very little posts mentioning them.

![title](output/DogStageMostRTFav.png)

# Analysis by Weekday

In [4]:
weekday_stats_df

count  count_per  favorite_count_sum  retweet_count_sum  \
Sunday       274  12.791783             2371771             723797   
Monday       356  16.619981             3044433             953858   
Tuesday      319  14.892624             2845248             863390   
Wednesday    319  14.892624             3197782            1023370   
Thursday     302  14.098973             2506778             772863   
Friday       298  13.912232             2582356             825072   
Saturday     274  12.791783             2442530             799009   

           rating_num_mean  rating_den_mean  favorite_count_mean  \
Sunday           13.007299        10.912409          8656.098540   
Monday           15.452247        10.112360          8551.778090   
Tuesday          11.454545        10.689655          8919.272727   
Wednesday        10.626959        10.094044         10024.394984   
Thursday         11.503311        10.463576          8300.589404   
Friday           12.181208        11.161074          8665.624161   
Saturday         10.686131        10.000000          8914.343066   

           retweet_count_mean  
Sunday            2641.594891  
Monday            2679.376404  
Tuesday           2706.551724  
Wednesday         3208.056426  
Thursday          2559.149007  
Friday            2768.697987  
Saturday          2916.091241

Let us now focus on each particular day of the week and its influence on our data. Bellow is shown that bar graph for each day of the week and a respective statistic. On the first set we can infer the following:

- The owner of the page posts the most on Mondays. 16.6% of the posts occur on a Monday versus 12.8% on Sunday and Saturday, the day he posts the least.

- People tend to like (Favorite) his posts more on Wednesdays.

- People Retweet his posts more on Wednesdays

One can say that users tend to like/retweet his posts with two days lag or that posting more does not grant an increased activity in Favorites and Reteweets.

![title](output/WeekdayCountRTFav.png)

Furthering the analysis by weekday, please refer to the plots bellow:

- The average Favorite and Retweet (Average per Tweet) are higher on Wednesdays. This serves as extra confirmation of what was mentioned above.

- The author of the page tends to give better ratings to dogs on Mondays, on average. 

![title](output/WeekdayMeanCountRTFav.png)

# Best Tweets

## Most Favorited Tweet

The most favorited Tweet was of a Lakeland Terrier, Puppo stage. Analysing the tweet, one can explain why this got the most Favorites. This was tweeted one day after President Trump assumed the Presidency, on Jannuary 21st 2017, during a #Womensmarch. This was a highly controversial period when people turned to the Social Media and were highly active with the use of popular hashtags.

Refer to the table bellow with the Top 5 Most Favorited Tweets.

In [5]:
twitter_archive_master.sort_values(by=['favorite_count'], ascending = False).head()[['created_at','favorite_count','retweet_count','name','url','rating_num','dog_stage','breed' ]]

created_at  favorite_count  retweet_count         name  \
331  2017-01-21 18:26:02          131574          47597  Not Defined   
838  2016-06-18 18:26:18          129874          78618  Not Defined   
427  2016-12-09 06:17:20          124066          62106      Stephan   
60   2017-06-26 19:07:24          107226          45362      Duddles   
115  2017-05-22 00:28:40          105941          32475       Jamesy   

                         url  rating_num    dog_stage               breed  
331  https://t.co/nTz3FtorBc          13        puppo    Lakeland Terrier  
838  https://t.co/7wE9LTEXC4          13        doggo  Labrador Retriever  
427  https://t.co/DkBYaCAg2d          13  not_defined           Chihuahua  
60   https://t.co/UDT7ZkcTgY          13  not_defined    English Springer  
115  https://t.co/wk7TfysWHr          13       pupper      French Bulldog

## Most Retweeted Tweet

The most Retweeted was a video of an adorable Labrador Retriever in the Doggo Stage. This tweet did not use any special hashtags nor was posted during a particular special time. This proves that Tweets can go viral with good content and quality, such is in this video, completely worth watching it.

Refer to the table bellow with the Top 5 Most Retweeted Tweets.

In [6]:
twitter_archive_master.sort_values(by=['retweet_count'], ascending = False).head()[['created_at','favorite_count','retweet_count','name','url','rating_num','dog_stage','breed' ]]

created_at  favorite_count  retweet_count         name  \
838  2016-06-18 18:26:18          129874          78618  Not Defined   
427  2016-12-09 06:17:20          124066          62106      Stephan   
877  2016-06-04 23:31:25           74352          51738  Not Defined   
331  2017-01-21 18:26:02          131574          47597  Not Defined   
60   2017-06-26 19:07:24          107226          45362      Duddles   

                         url  rating_num    dog_stage               breed  
838  https://t.co/7wE9LTEXC4          13        doggo  Labrador Retriever  
427  https://t.co/DkBYaCAg2d          13  not_defined           Chihuahua  
877  https://t.co/YcXgHfp1EC          13        doggo          Eskimo Dog  
331  https://t.co/nTz3FtorBc          13        puppo    Lakeland Terrier  
60   https://t.co/UDT7ZkcTgY          13  not_defined    English Springer

# Final Remarks

After this careful and extensive analysis, we can offer the following insights to increase the odds to the users that pretend to be featured on the WeRateDogs page and have his dog seen by the world:

    - Having a Golden Retriever or Labrador Retriever in the Pupper stage
    - On Mondays the Author will Tweet the most
    - Tweets are more Favorited/Retweeted on Wednesdays
    - Author tends to give better Ratings on Mondays
    - Having a French Bulldog seems to be popular amongst Twitter Users
    - Puppos tend to be Favorited/Retweeted the most on average by Users
    - Use of particular hashtags that are viral due to the circumstance of the season